packages/blitz/src/middleware.ts

```diff
@@ -27,8 +27,8 @@ export async function handleRequestWithMiddleware<
 ) {
   if (!(res as unknown as MiddlewareResponse).blitzCtx) {
     ;(res as unknown as MiddlewareResponse).blitzCtx = {
-      revalidatePage: revalidateFactory(res as unknown as NextApiResponse),
-    } as Ctx
+      revalidatePage: revalidateFn(res as unknown as NextApiResponse),
```

Comment:
Why are we assuming that `res` is assignable to the NextApiResponse type and has `revalidate` property?

In the handleRequestWithMiddleware, the response is typed as ServerResponse. It can be overwritten and extended, but the minimal accepted type is ServerResponse. Hence I don't think we can assume that it's NextApiResponse. Especially since this is `blitz` package which should remain next-agnostic (that's why we have own types, and we're not using Next's types anywhere in this file.

Reply:
One solution that I can see here is moving this logic to blitz-next — i.e. having a middleware there.

Reply:
Added to blitz-next